src/MonoTorrent/MonoTorrent.Client.Encryption/EncryptionType.cs

```diff
@@ -54,12 +54,14 @@ internal static bool SupportsRC4 (IList<EncryptionType> allowedEncryption)
 
         internal static IList<EncryptionType> Remove (IList<EncryptionType> allowedEncryption, EncryptionType encryptionType)
         {
-            var result = new EncryptionType[allowedEncryption.Count - 1];
-            int j = 0;
-            for (int i = 0; i < allowedEncryption.Count; i++)
-                if (allowedEncryption[i] != encryptionType)
-                    result[j++] = allowedEncryption[i];
-
+            var result = allowedEncryption;
+            if (result.Count > 0) {
```

Comment:
While writing a test I realised that this introduced an unexpected change.

```
var result = EncryptionTypes.Remove (new List<EncryptionType> (), EncryptionType.PlainText);
result.Add (EncryptionType.PlainText);
```

This should throw an exception as `result` should be immutable. With your initial change this was no longer true. I tweaked it to always return an immutable empty collection when the input list is empty!

Reply:
Oh derp, my comment is incorrect - the original code was fine in that respect. It would still have been wrapped with the `MakeReadOnly` at the end of the call. Never mind! That's my bad for mis-reading after the original review 😆